teerex/src/lib.rs

```diff
@@ -359,9 +365,13 @@ pub mod pallet {
 			log::info!("teerex: called into runtime call register_quoting_enclave()");
 			// Quoting enclaves are registered globally and not for a specific sender
 			let _sender = ensure_signed(origin)?;
-			let quoting_enclave =
-				Self::verify_quoting_enclave(enclave_identity, signature, certificate_chain)?;
+			let quoting_enclave = Self::verify_quoting_enclave(
+				enclave_identity.clone(),
+				signature,
+				certificate_chain,
+			)?;
 			<QuotingEnclaveRegistry<T>>::put(quoting_enclave);
+			Self::deposit_event(Event::QuotingEnclaveRegistered { enclave_identity });
```

Comment:
Hmm, sorry. I think I have approved to fast. I have been pondering about this, and I believe we should actually put the whole quoting enclave struct here instead of the encoded vec.

Reply:
Will do